### DISCUSSION TITLE
Add more flexibility about the twig version supported.

### DISCUSSION THREAD

**Discussion by VincentLanglet** [craftcms/cms#15568] [Q&A] [answered]
Hi @brandonkelly 

I'm writing a tool https://github.com/VincentLanglet/Twig-CS-Fixer which require twig 3.10 and got recently two issues with people using craftcms because it use twig 3.8.

When looking at the twig version in this repository:
- The minor version is locked https://github.com/craftcms/cms/blob/cd245dfc070727a80c41e26b91a038a2121c4c34/composer.json#L64
- Last bump was done months ago https://github.com/craftcms/cms/pull/14470

Since minor version doesn't introduce BC break I would like to know why `^3.8` is not used to declare the twig dependencies ?
Also, if `~` want to be kept ; would it be possible to bump again the twig version to something equal or bigger than 3.10 ?

Thanks a lot.

**Comment by brandonkelly** [comment, marked as answer]
There are [breaking changes](https://github.com/twigphp/Twig/issues/4146) in Twig 3.9+ which we’re waiting on fixes for, before we can update Craft past 3.8. (The issue is closed and partially fixed for 3.12, but not fully.)

**Reply by brandonkelly** [reply]
Craft 4.12.0 and 5.4.0 are out now with Twig 3.12.
